src/main/java/org/montrealjug/billetterie/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.montrealjug.billetterie.security;
+
+import jakarta.servlet.SessionTrackingMode;
+import java.util.Set;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.server.Cookie;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration(proxyBeanMethods = false)
+@EnableWebSecurity
+@EnableConfigurationProperties(SecurityConfiguration.AdminProperties.class)
+public class SecurityConfiguration {
+
+    private static final String ADMIN_ROLE = "admin";
+    private static final String ACTUATOR_ROLE = "actuator";
+
+    // we define the basic auth for actuator before
+    // to avoid clash between our security chains, we have to use `securityMatcher`
+    // to scope this security chain on the `actuator/**` path only.
+    // as `actuator` is `read-only` in the config, we have a small risk of csrf:
+    // an attacker could forge a link that, if clicked by a user with saved `actuator`
+    // credentials in their browser, could leak some info.
+    // as `threaddump` and `heapdump` are also disabled, I think we can live
+    // with the threat of leaking our `beans` hierarchy or our `loggers` config
+    @Bean
+    @Order(0)
+    SecurityFilterChain actuatorBasicAuth(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity
+                .securityMatchers(
+                        securityMatcher -> securityMatcher.requestMatchers("/actuator/**"))
```

Comment:
actually, I don't think it is a good idea to completely hide the actuator behind an auth wall.

I'm thinking about `/actuator/health` this should be public I believe, so that I could use it from fly.io  or any other uptime service for example.

the only "sensitive" info in `/actuator/health` is a file path about disk storage, and I'm pretty sure we don't care

WDYT?

Reply:
ok, good idea, let's just add probes and be done with it.

you can add some fly config, I'll test it out after it's merged and fix it if needed

Thanks!